src/service/video-manager-impl.js

```diff
@@ -437,6 +556,216 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed (at most once per
+   * animation frame)
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   */
+  onDockableVideoPositionChanged_(newPos) {
+
+
+    const inViewportHeight = this.calculateDockableVideoPosition_(newPos);
+
+    // Short-circuit the position change handler if the video isn't loaded yet
+    // or is not playing manually while in-line (paused videos need to go
+    // through if they are docked since this method handles the "undocking"
+    // animation)
+    if (!this.loaded_
+      || !this.internalElement_
```

Comment:
based on suggestions above, if we cache rect and use vsyn, need to add `|| !this.initialRect_` as well in case we get here before vsyn has measured it ( `!this.loaded_` probably protects against this ever happening, but it is safer code)